pkg/machines/components/networks/utils.js

```diff
@@ -188,15 +188,28 @@ export function validateIpv6Prefix(prefix) {
  * @param {string} ip
  * @returns {string}
  */
-export function ipv6ToBinStr(ip) {
-    const parts = [];
-    ip.split(":").forEach(part => {
-        let bin = parseInt(part, 16).toString(2);
+function ipv6ToBinStr(ip) {
+    const validGroupCount = 8;
+    /* Split address by `:`
+     * Then check if the array contains an empty string (happens at ::), and if so
+     * replace it with the appropriate number of 0 entries.
+     */
+    const arrAddr = ip.split(":").reduce((accum, hexNum) => {
+        if (hexNum)
+            accum.push(hexNum);
+        else
+            for (let i = 0; i < (validGroupCount - ip.split(":").length + 1); i++)
```

Comment:
This splits `ip` twice -- better do it once and put that into a `const`.

Reply:
Fixed.